.github/workflows/branch-preview.yaml

```diff
@@ -0,0 +1,31 @@
+name: Branch Preview
+
+on: workflow_dispatch
+
+jobs:
+    main:
+        name: Build and Deploy
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v3
+              with:
+                  ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
https://github.com/actions/checkout?tab=readme-ov-file#checkout-pull-request-head-commit-instead-of-merge-commit